entitlement/src/main/java/org/killbill/billing/entitlement/api/DefaultBaseEntitlementWithAddOnsSpecifier.java

```diff
@@ -48,11 +60,19 @@ public UUID getBundleId() {
         return bundleId;
     }
 
+    public void setBundleId(final UUID bundleId) {
```

Comment:
Mixed reviews... the object was immutable, now it's not. I know it's convenient, but i am wondering if creating another object would not be more maintainable. I am thinking all the payment context for which we have setters, and it's hard to follow the code. if you want to keep it this way, not a strong objection either.

Reply:
Unfortunately, all our code is structure around:

```
public interface WithEntitlementPlugin<T> {
    T doCall(final EntitlementApi entitlementApi, final DefaultEntitlementContext updatedPluginContext) throws EntitlementApiException;
}
```

`T` could be `UUID`, `List<UUID>`, etc. So the context has to be mutable (passed in as argument).